File: Challenges/Problem_4/Largest_palindrome_product.js
// KuntaNgcobo
// Largest palindrome product

const isPalindrome = n => {
  let stringN = n + '';
  let length = stringN.length;
  let minLength = length%2;

  while(stringN.length > minLength){
    let start = stringN.slice(0,1);
    let end = stringN.slice(-1);
    if(start !== end){
      return false;
    }
    stringN = stringN.slice(1, -1)
  }
  return true;
}

const largestPalindromeProduct = (n) => {
  let start = Math.pow(10, n-1) - 1;
  let end = Math.pow(10, n) - 1;
  let largest = [0, 0, 0];
  let i = end;
  let j = end;
  while(i > start){
    while (j > start){
      let number = i*j;
      if(largest[2] + largest[1] < i+j && isPalindrome(number)){
              largest = [number, i, j];
              //break loopj;
              j=start;
      }else{
        j--;
      }
    }
    j=end;
    i--;
    if(i <= largest[2]){
      break;
    }
  }
  return largest[0];
}

File: Challenges/Problem_4/Largest_palindrome_product.py
# KuntaNgcobo
# Largest palindrome product

import math

def isPalindrome(n):
  stringN = str(n)
  return stringN == stringN[::-1]


def largestPalindromeProduct(n):
  start = pow(10, n-1) - 1
  end = pow(10, n) - 1
  largest = [0, 0, 0]
  i = end
  j = end
  while(i > start):
    while (j > start):
      number = i*j
      if(((largest[2] + largest[1]) < (i+j)) and isPalindrome(number)):
              largest = [number, i, j]
              j=start
      else:
        j=j-1

    j=end
    i=i-1
    if(i <= largest[2]):
      break
  #print(largest)
  return largest[0]
